### DISCUSSION TITLE
`navigateTo` in nuxt router middleware

### DISCUSSION THREAD

**Discussion by undermoonn** [nuxt/nuxt#16124] [Questions]
I try to use `navigateTo` in nuxt route middleware to redirect request to a new path, and I expected that page component only run after redirection. But page component runs twice.

Here are my codes:
`middleware/redirect.global.ts`

```ts
export default defineNuxtRouteMiddleware((to) => {
  if (to.path === '/app/1') {
    return navigateTo('/app/fortnite-1', { redirectCode: 301 });
  }
})
```

`pages/app/[id].vue`
```vue
<script lang="ts" setup>
console.log('run component') // `run component` log twice
</script>

<template>
  <h1>app</h1>
</template>

```

Can I prevent from this? Or maybe there is a better way of redirection 🤔

**Comment by duc2019** [comment]
Check workround here:
https://github.com/nuxt/nuxt/issues/14468

Similar issue:
https://github.com/nuxt/nuxt/discussions/16968

This is still not fixed in nuxt 3.2.2, atleast for project with ssr disabled.
The workround above will show 404 before redirect, this is what I used (it's ugly but it works very well)

``` javascript
location.replace(path)
await new Promise(resolve => setTimeout(resolve, 99999)) 
```
